src/engine/server_iv.c

```diff
@@ -182,23 +182,26 @@ ds_iv_ns_put(struct ds_iv_ns *ns)
 		ds_iv_ns_destroy(ns);
 }
 
-static struct ds_iv_ns *
-iv_ns_lookup_by_ivns(crt_iv_namespace_t ivns)
+static int
+iv_ns_lookup_by_ivns(crt_iv_namespace_t ivns, struct ds_iv_ns **p_ns)
 {
 	struct ds_iv_ns *ns;
 
+	*p_ns = NULL;
 	d_list_for_each_entry(ns, &ds_iv_ns_list, iv_ns_link) {
-		if (ns->iv_ns == ivns) {
-			if (!ns->iv_stop) {
-				ds_iv_ns_get(ns);
-				return ns;
-			}
+		if (ns->iv_ns != ivns)
+			continue;
+
+		if (ns->iv_stop) {
 			D_DEBUG(DB_MD, DF_UUID" stopping\n",
 				DP_UUID(ns->iv_pool_uuid));
-			return NULL;
+			return -DER_SHUTDOWN;
```

Comment:
Sending -DER_SHUTDOWN from one rank to another sounds strange, if not potentially confusing. It is used to indicate that, locally, a lower layer has encountered an error (e.g., ENOMEM, EIO), and upper layers of the service (e.g., Pool Service) should trigger a shutdown, so that a different service replica could take over. In this PR's case, "the IV NS is shutting down" should not imply that "a different rank should shut down", right?

Reply:
TBH, I do not see much of confusion to use SHUTDOWN outside of service change. To me shutdown just means the service is being shutdown somewhere, either switch/ignore/failure depend on the caller.

I am not sure we need invite a new error code here, what error do you suggest?

Reply:
@liw  any error code suggestion?